src/main/java/de/siegmar/logbackgelf/SimpleJsonEncoder.java

```diff
@@ -107,36 +107,50 @@ private void appendKey(final String key) throws IOException {
      */
     @SuppressWarnings("checkstyle:cyclomaticcomplexity")
     private void escapeString(final String str) throws IOException {
-        for (final char ch : str.toCharArray()) {
+        int currentIndex = 0;
+        int lastCopyIndex = 0;
+        char[] chars = str.toCharArray();
+        for (final char ch : chars) {
+            String escaped = null;
             switch (ch) {
```

Comment:
It might be beneficial to refactor the switch block to a private static method and call it via `final String escaped = escapeIfNeeded(ch);`. Another possible optimization would be to have `private static final char` attributes for all the escape sequences to test for.

Reply:
I'm going to work on a way to escape the string only if is needed. 

I did some tests putting some values on `static final` variables, but it didn't show a significant difference